discord/application.go

```diff
@@ -114,11 +144,37 @@ const (
 	NumberOption
 )
 
-type CommandOptionChoice struct {
+type CommandOptionChoice interface {
+	Type() CommandOptionType
```

Comment:
Make this `optType() CommandOptionType` so the user can't implement their own types, because it would be pointless and erroneous to do so.